inbm/dispatcher-agent/dispatcher/schedule/apscheduler.py

```diff
@@ -0,0 +1,45 @@
+"""
+    Uses APScheduler to execute scheduled tasks.
+
+    Copyright (C) 2024 Intel Corporation
+    SPDX-License-Identifier: Apache-2.0
+"""
+
+import logging
+from typing import Callable
+
+from .schedules import SingleSchedule, RepeatedSchedule
+from apscheduler.schedulers.background import BackgroundScheduler
+
+logger = logging.getLogger(__name__)
+
+
+class APScheduler:
+    def __init__(self) -> None:
+        self._scheduler = BackgroundScheduler()
+
+    def start(self) -> None:
+        """Start the scheduler"""
+        logger.debug("")
+        self._scheduler.start()
```

Comment:
**<h2>$\color{#15F4EE}Miscellaneous$</h2>**
**Suggestion:** Ensure that the scheduler is properly shutdown to release resources.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    def start(self) -> None:
        """Start the scheduler"""
        logger.debug("Starting the APScheduler.")
        self._scheduler.start()

    def shutdown(self) -> None:
        """Shutdown the scheduler"""
        logger.debug("Shutting down the APScheduler.")
        self._scheduler.shutdown()
```

Reply:
On this one, I wonder if we call shutdown that it kills all the jobs.  Maybe we could use that when we want to flush all of the jobs out of the schedule when we receive a new manifest.